examples/pennylane_run_quantum_natural_gradient.py

```diff
@@ -0,0 +1,451 @@
+    return var(qml.PauliY(1)), var(qml.PauliX(2))
+
+##############################################################################
+# As previously, the diagonal terms are simply given by the variance,
+
+varK0, varK1 = layer2_diag(params)
+g2[0, 0] = varK0/4
+g2[1, 1] = varK1/4
+
+
+##############################################################################
+# while the off-diagonal terms require covariance between the two
+# observables to be computed.
+
+@qml.qnode(dev)
+def layer2_off_diag_single(params):
+    layer2_subcircuit(params)
+    return expval(qml.PauliY(1)), expval(qml.PauliX(2))
+
+@qml.qnode(dev)
+def layer2_off_diag_double(params):
+    layer2_subcircuit(params)
+    X =  np.array([[0, 1], [1, 0]])
+    Y =  np.array([[0, -1j], [1j, 0]])
+    YX = np.kron(Y, X)
+    return expval(qml.Hermitian(YX, wires=[1, 2]))
+
+# calculate the off-diagonal terms
+exK0, exK1 = layer2_off_diag_single(params)
+exK0K1 = layer2_off_diag_double(params)
+
+g2[0, 1] = (exK0K1 - exK0*exK1)/4
+g2[1, 0] = g2[0, 1]
+
+
+##############################################################################
+# Putting this altogether, the block-diagonal approximation to the Fubini-Study
+# metric tensor for this variational quantum circuit is
+from scipy.linalg import block_diag
+g = block_diag(g1, g2)
+print(np.round(g, 8))
+
+
+##############################################################################
+# PennyLane contains a built-in method for computing the Fubini-Study metric
+# tensor, :meth:`.QNode.metric_tensor`, which we can use to verify this
+# result:
+print(np.round(circuit.metric_tensor(params), 8))
+
+##############################################################################
+# As opposed to our manual computation, which required 6 different quantum
+# evaluations, the PennyLane Fubini-Study metric tensor implementation
+# requires only 2 quantum evaluations, one per layer. This is done by
+# automatically detecting the layer structure, and noting that every
+# observable that must be measured commutes, allowing for simultaneous measurement.
+#
+# Therefore, by combining the quantum natural gradient optimizer with the analytic
+# parameter-shift rule to optimize a variational circuit with :math:`d` parameters
+# and :math:`L` parametrized layers, a total of :math:`2d+L` quantum evaluations
+# are required per optimization step.
+#
+# Note that the :meth:`.QNode.metric_tensor` method also supports computing the diagonal
+# approximation to the metric tensor:
+print(circuit.metric_tensor(params, diag_approx=True))
+
+##############################################################################
+# Quantum natural gradient optimization
+# -------------------------------------
+#
+# PennyLane provides an implementation of the quantum natural gradient
+# optimizer, :class:`~.QNGOptimizer`. Let's compare the optimization convergence
+# of QNG Optimizer and :class:~.GradientDescentOptimizer for the simple variational

```

Comment:
```suggestion
# of the QNG Optimizer and the :class:~.GradientDescentOptimizer for the simple variational
```

Reply:
Note: `:class:~.GradientDescentOptimizer` did not render correctly in docs built on my machine

Reply:
```suggestion
# of the QNG Optimizer and the :class:`~.GradientDescentOptimizer` for the simple variational
```
Fixed, it was missing the backticks